tools/coqdep_common.ml

```diff
@@ -280,6 +284,25 @@ let string_of_dependency_list suffix_for_require deps =
     in
   String.concat " " (List.map string_of_dep deps)
 
+let rec phys_path_best_match (prefix: string list) (logpath: string list) :  (string (* physical path corresponding to a prefix of logpath *) * string list (* unmatched suffix *)) option =
+  match logpath with
+  | [] -> (match find_physpath prefix with
+          | None -> None
+          | Some p -> Some (p,[]))
+  | h::tl ->
+          match phys_path_best_match (prefix@[h]) tl with
+          | None -> (match find_physpath prefix with
+                    | None -> None
+                    | Some p -> Some (p,h::tl)
+                    )
+          | Some p -> Some p
+
+let phys_path (logpath: string list) : string =
+  let (ppath, suffix) = match phys_path_best_match [] logpath with
+                        | None  -> ("", logpath)
+                        | Some x -> x in
+  String.concat "" (ppath::[String.concat "/" (""::suffix)])
```

Comment:
On Windows, we need to replace "/" by "\"? Is there a platform-independent function to get the filename separator?

Reply:
Windows accepts both `/` and `\`.  I'm not sure whether or not `make` freely converts between them, but if it does not, all you're responsible for here is making sure that you're consistent with the rest of the makefile generation process.